src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -187,9 +185,14 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         if ('' === $url) {
             $url = '/';
-        } elseif (preg_match($this->urlEncodingSkipRegexp, $url)) {
-            // the context base URL is already encoded (see Symfony\Component\HttpFoundation\Request)
+        } elseif (!empty($this->decodedChars)) {
+            @trigger_error('The class variable '.__CLASS__.'::$decodedChars is deprecated since version 2.7 and will be removed in 3.1.', E_USER_DEPRECATED);
```

Comment:
Please be more clear in what I'm supposed to do. This is basically my first PR.
I am not an experienced Symfony developer, and have no clear idea what I'm supposed to do or where to find the appropriate documentation. Only telling me what I'm doing wrong, won't improve this PR :/

I'd gladly donate this change/PR to someone who actually knows what to do. I just want the route-generation to be faster, since it takes a fairly large part of the symfony 'overhead'* of our application.

* Its not really overhead, but due to its flexibility it is much more complex than simply concatenating the right parts of a url.